src/Streamly/Internal/Data/Fold/Channel/Type.hs

```diff
@@ -510,10 +510,12 @@ sendToWorker_ chan a = go
                     (inputItemDoorBell chan)
                     (ChildYield a)
         else do
-            error "sendToWorker_: No space available in the buffer"
+            -- XXX: Is there a reason for us to error out here instead of
+            -- blocking?
```

Comment:
this might have been introduced to know that we are actually hitting this condition instead of the possibility of a deadlock due to blocking. If there is a possibility of deadlock then you better ensure that the buffer is always available.